rotkehlchen/chain/evm/decoding/balancer/utils.py

```diff
@@ -0,0 +1,201 @@
+                protocol=CPT_BALANCER_V1,
+                name=pool['name'].strip(),
+                symbol=pool['symbol'].strip(),
+                decimals=deserialize_int(pool['decimals']),
+                evm_address=string_to_evm_address(to_checksum_address(pool['address'])),
+                encounter=TokenEncounterInfo(
+                    description=f'Querying {CPT_BALANCER_V1} balances',
+                    should_notify=False,
+                ),
+                underlying_tokens=[
+                    UnderlyingToken(
+                        address=get_or_create_evm_token(
+                            userdb=inquirer.database,
+                            chain_id=inquirer.chain_id,
+                            protocol=CPT_BALANCER_V1,
+                            name=token['name'].strip(),
+                            symbol=token['symbol'].strip(),
+                            evm_address=string_to_evm_address(to_checksum_address(token['address'])),
+                            encounter=TokenEncounterInfo(
+                                description=f'Querying {CPT_BALANCER_V1} balances',
+                                should_notify=False,
+                            ),
+                        ).evm_address,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=FVal(token['weight']),
+                    )
+                    for token in pool['displayTokens']
+                ],
+            )
+            if pool_token.protocol != CPT_BALANCER_V1:
+                log.debug(f'Updating protocol for balancer asset {pool_token}')
+                GlobalDBHandler.set_token_protocol_if_missing(
+                    token=pool_token,
+                    new_protocol=CPT_BALANCER_V1,
+                )
+        except (KeyError, ValueError, TypeError, DeserializationError) as e:
+            log.error(
+                f'Failed to process balancer v1 pool due to {e!s}. '
+                f'Pool: {pool}. Skipping...',
+            )
+            continue
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_unique_cache_value_at_ts(
+            write_cursor=write_cursor,
+            key_parts=cache_key_parts,
+            value=str(latest_pools_count),
+            timestamp=now,
+        )
+
+
+def _query_balancer_pools_count(chain: 'ChainID', version: Literal[1, 2]) -> int:
+    """Fetch the total number of pools for the specified chain and protoco
+    May raise:
+        - RemoteError
+    """
+    try:
+        response = requests.post(
+            url=BALANCER_API_URL,
+            json={
+                'variables': {
+                    'chain': CHAIN_ID_TO_BALANCER_API_MAPPINGS[chain],
+                    'version': version,
+                },
+                'query': GET_POOLS_COUNT_QUERY,
+            },
+            timeout=CachedSettings().get_timeout_tuple(),
+        )
+        response.raise_for_status()
```

Comment:
mmmm that is not what you want I think? Does this raise if you receive a 404 error? have you checked it? also what are the responses from the api?